engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -152,6 +159,21 @@
     private DisplayResolutionDependentFBOs displayResolutionDependentFBOs;
     private ShadowMapResolutionDependentFBOs shadowMapResolutionDependentFBOs;
 
+    private static RenderGraph renderGraph = new RenderGraph(); // TODO: Try making this non-static
+
+    // Required for the reflection magic that Systems use.
```

Comment:
On point 1, I'd say let's have one or two methods to set only once those variables, immediately after instantiation. Everything else can wait postBegin().